src/app/app.component.ts

```diff
@@ -38,28 +44,30 @@ export class AppComponent implements AfterViewInit {
   REDIRECT_URI: string = 'REPLACE_WITH_YOUR_REDIRECT_URI'; // i.e http://localhost:3000, https://serverName/lf-sample/index.html
   CLIENT_ID: string = 'REPLACE_WITH_YOUR_CLIENT_ID';
   HOST_NAME: string = 'laserfiche.com'; // only update this if you are using a different region or environment (i.e. laserfiche.ca, eu.laserfiche.com)
-  REGIONAL_DOMAIN: string = 'laserfiche.com' // only update this if you are using a different region or environment
+  REGIONAL_DOMAIN: string = 'laserfiche.com'; // only update this if you are using a different region or environment
 
   // repository client that will be used to connect to the LF API
   private repoClient?: IRepositoryApiClientExInternal;
-
-  // url to open the selected node in Web Client
-  selectedNodeUrl?: string;
-
-  // the folder the user has selected in the folder-browser
-  selectedFolder?: LfFolder;
-
   // used to get the file user is trying to save
   @ViewChild('fileInput') fileInput: ElementRef<HTMLInputElement>;
 
+  lfSelectedFolder: ILfSelectedFolder;
+  lfFieldContainerElement: ElementRef<LfFieldContainerComponent>;
+
   // the UI components
-  @ViewChild('lfFieldContainerElement') lfFieldContainerElement?: ElementRef<LfFieldContainerComponent>;
+  @ViewChild('lfFieldContainerElement')  set lfFieldContainerContent(lfFieldContainerRef: ElementRef<LfFieldContainerComponent>) {
+    if (lfFieldContainerRef) {
+      this.lfFieldContainerElement = lfFieldContainerRef;
+      this.initializeFieldContainerAsync(); // TODO: cannot await this
```

Comment:
how many times does this get  called?

Reply:
I think we've solved a similar issue before by using hidden instead of ngIF?

Reply:
https://stackoverflow.com/questions/34947154/angular-2-viewchild-annotation-returns-undefined